### PR TITLE
Add day of event in footer and on dashboard

### PR DESCRIPTION
**Pull request by weeman1337**
[pipeline](https://chaos.expert/engelsystem/engelsystem/-/pipelines/39972)

image

### CODE REVIEW

File: tests/Unit/Helpers/DayOfEventTest.php [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+            'day -2 (23:59, with day 0)' => [-2, true,  Carbon::createFromFormat(self::FORMAT, '2023-07-28 23:59:59')],
+            'day -1 (23:59, with day 0)' => [-1, true,  Carbon::createFromFormat(self::FORMAT, '2023-07-29 23:59:59')],
+            'day 0 (with day 0)'        => [0, true, Carbon::createFromFormat(self::FORMAT, '2023-07-30 16:00:00')],
+            'day 1 (with day 0)'         => [1, true, Carbon::createFromFormat(self::FORMAT, '2023-07-31 10:00:00')],
+            'day 2 (with day 0)'         => [2, true, Carbon::createFromFormat(self::FORMAT, '2023-08-01 10:00:00')],
+
+            'day -2 (without day 0)'        => [
+                -2,
+                false,
+                Carbon::createFromFormat(self::FORMAT, '2023-07-29 10:00:00'),
```
Comment (MyIgel) [inline tests/Unit/Helpers/DayOfEventTest.php:49]: In general i would put the `Carbon::createFromFormat(self::FORMAT, [...]` inside the test method, that would make it easier to grasp

**Review by MyIgel — CHANGES_REQUESTED** [review]
There seems to be a bug in the calculation, for example with the current CCCAMP23 config it shows day 0 instead of -2

File: src/Helpers/DayOfEvent.php [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+
+        if (!$startOfEvent) {
+            return null;
+        }
+
+        /** @var Carbon $startOfEvent */
+        $startOfEvent = $startOfEvent->copy()->startOfDay();
+
+        $now = Carbon::now()->startOfDay();
+        $diff = $now->diffInDays($startOfEvent, false);
```
[the remark this answers is not in the document]
Comment (weeman1337) [reply src/Helpers/DayOfEvent.php:29]: The diff should be the way around. Tests were crap as well.
Probably fiddling around without enough sleep is a bad idea :laughing:

File: tests/Unit/Helpers/DayOfEventTest.php [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+     */
+    public function provideTestGetData(): array
+    {
+        return [
+            'no start date' => [null, true, null],
+
+            'day -2 (10:00, with day 0)' => [-2, true, Carbon::createFromFormat(self::FORMAT, '2023-07-28 10:00:00')],
+            'day -2 (23:59, with day 0)' => [-2, true,  Carbon::createFromFormat(self::FORMAT, '2023-07-28 23:59:59')],
+            'day -1 (23:59, with day 0)' => [-1, true,  Carbon::createFromFormat(self::FORMAT, '2023-07-29 23:59:59')],
+            'day 0 (with day 0)'        => [0, true, Carbon::createFromFormat(self::FORMAT, '2023-07-30 16:00:00')],
```
[the remark this answers is not in the document]
Comment (weeman1337) [reply tests/Unit/Helpers/DayOfEventTest.php:42]: We should start using PHP-CS-Fixer
https://github.com/engelsystem/engelsystem/issues/1187

File: tests/Unit/Helpers/DayOfEventTest.php (hunk 1 above)
Comment (weeman1337) [reply tests/Unit/Helpers/DayOfEventTest.php:49]: Done

**Pull request merged**